sdk/src/openssl/ffi_mutex.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+// OpenSSL code is not re-entrant. Use this to guard against race conditions.
+use crate::Result;
+
+#[cfg(feature = "openssl_ffi_mutex")]
+static FFI_MUTEX: std::sync::Mutex<()> = std::sync::Mutex::new(());
+
+pub(crate) struct OpenSslMutex<'a> {
+    // Dead code here is intentional. We don't need to read the () contents
+    // of this guard. We only need to ensure that the guard is dropped when
+    // this struct is dropped.
+    #[allow(dead_code)]
+    #[cfg(feature = "openssl_ffi_mutex")]
```

Comment:
If we want this to be an optional feature, then why not declare it where we use it instead of having a dummy implementation and always calling it?

Alternatively we could isolate the openssl methods we call into a wrapper that does the right checks when needed.

Reply:
I think the goal is to have this always enabled in all builds, so ok to continue as is, and we will remove the feature ASAP when tested.